src/core/tsi/transport_security_interface.h

```diff
@@ -258,6 +258,12 @@ typedef struct tsi_handshaker_result tsi_handshaker_result;
 tsi_result tsi_handshaker_result_extract_peer(const tsi_handshaker_result* self,
                                               tsi_peer* peer);
 
+// This method extracts tsi local peer. It returns TSI_OK assuming there is no fatal
+// error.
+// The caller is responsible for destructing the local peer.
+tsi_result tsi_handshaker_result_extract_local_peer(const tsi_handshaker_result* self,
```

Comment:
I think it may be overkill to add a new API to TSI - instead, can we just add the properties to the `tsi_peer` that we build during `tsi_handshaker_result_extract_peer`?

There is precedent for having local connection info in the `tsi_peer` (e.g. see the ALTS TSI handshaker) and it has several advantages:
1. Much less work in this PR and to import the code (there are many internal TSI implementations that would need updating).
2. Easier to maintain, because we don't have 2 parallel but nearly identical flows.

Reply:
Just to add an additional comment, I find this phrase `local_peer` confusing. IIUC, `peer` in this context usually means the other side.